### ISSUE TITLE
"Un-tilt" photos in web editor

### ISSUE THREAD

**Issue by pbb72** [mapillary/mapillary_issues#770]
I am making most photos while walking, and I cannot watch the screen constantly. (If I make photos sideways, I cannot see the screen at all.) Because of this, lots of my photos are slightly tilted. That is, the horizon is lopsided by 5-10 degrees. This looks very amateurish, even if the rest of the photo is great. See for example: http://www.mapillary.com/map/im/MCoYgzFPS7gKXagXVSHC_A

It would be great if I could correct the tilting afterwards in the web editor. Cut off a little bit from all sides, and the corrected photo can keep the same aspect ration.

**Comment by ybkuang** [comment]
Yes! We are thinking about making this possible together with rotation editing in the next version of the the editor.

**Comment by de-vries** [comment]
How hard would it be to do this automatically? Or even semi-automatically, you could build a game for that ;)

**Comment by ybkuang** [comment]
;) We tried one idea long time ago and noticed that we can correct most of tilts automatically. But the main problem is that we also tilt some of the good images. One thing we can test is that we can feed the automatic tilt correction to the editor UI and let the user help deciding whether it is good or not (semi-auto).

**Comment by polarbearing** [comment]
Be careful with automated untilting. 

The side acceleration when driving in a curve is similar to a tilt, and would lead to false positives.

**Comment by pbb72** [comment]
I think the best way with automatic correction is to offer a suggestion in the photo edit screen. Like "This photo seems to be tilted. Does this correction improve the image?", together with buttons: "No", "Yes", "Manually correct" and "Automatically correct whole sequence", and a before and after preview side-by-side. An undo button should be available for each individual image, in case the automatic correct did not work well for every image.

Looking at my photos, I can easily have several hundreds of photos per week that need to be corrected. If I would get a list of all photos that get automatic correction, then I would quickly start to ignore those lists, so it would be of very little use.

**Comment by de-vries** [comment]
^ this :+1:

**Comment by nighto** [comment]
Can I second this request? It'd improve Mapillary a lot. Could be done on the web (post-upload) or on the app (pre-upload). Thanks.

**Comment by nighto** [comment]
A year later, any thoughts on this issue? Sorry to just add noise, but I thing this would improve Mapillary a lot. Could even have a game to fine tune the algorithm.

**Comment by pbb72** [comment]
@nighto, for the last few months, no editing has been possible at all (see #1736 and #1759), so I think you will need some more patience for new functionality :-)

**Comment by ybkuang** [comment]
mapillary_issues is no longer an active repository.
Let’s continue discussing features in our forum! We will import this issue as topic to the Mapillary Forum in the coming weeks. We’ll notify here once it’s been done. See you there!
